src/core/thread/data_poll_manager.cpp

```diff
@@ -351,6 +363,23 @@ void DataPollManager::SendFastPolls(uint8_t aNumFastPolls)
     }
 }
 
+otError DataPollManager::StopFastPolls(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mFastPollsUsers != 0);
+
+    mFastPollsUsers--;
```

Comment:
How about adding this before `mFastPollsUser--`?
```
VerifyOrExit(mFastPollsUsers < kMaxFastPollsUsers);
```

To cover this case
- In `SendFastPolls()` we don't let the `mFastPollsUser` to be incremented after we get to max `kMaxFastPollsUsers
- If we hit max, and still allow it to be decremented from `StopFastPollsUsers` the counter may become incorrectly zero.
- When `mFastPollsUser` is at max, we don't decrement it anymore and let it be cleared from `HandlePollSent()` (after all fast polls are sent).